crates/bvs-vault-bank/src/contract.rs

```diff
@@ -269,6 +273,37 @@ mod execute {
             )
             .add_message(send_msg))
     }
+
+    /// transfer the assets to the router as part of slashing flow.
+    /// The idea is in the event of slashing router will maintain custody of the assets.
+    /// Whether slashed collateral is burned, further utilized or returned to the user is up to the
+    /// downstream slashing logics router will be working with.
+    pub fn system_lock_asset(
+        deps: DepsMut,
+        env: Env,
+        info: MessageInfo,
+        amount: Amount,
+    ) -> Result<Response, ContractError> {
+        router::assert_router(deps.as_ref().storage, &info)?;
+
+        // if the code get passed above assert_router, it means the sender is the router
+        // No need to load from storage.
+        let router = info.sender;
+
+        let vault_balance = bank::query_balance(&deps.as_ref(), &env)?;
+
+        if amount.0 > vault_balance {
+            return Err(VaultError::insufficient("Not enough balance").into());
+        }
+
+        let transfer_msg = bank::bank_send(deps.storage, &router, amount.0)?;
+
+        let event = Event::new("SystemLockAssets")
```

Comment:
nit: since fn is singular "asset", should standardise? 
```suggestion
        let event = Event::new("SystemLockAsset")
```

Reply:
@fuxingloh Notice you changed the PR title to slash_locked , would like a say in name picking of actual execute msg?.

Reply:
I think let's call it `SlashLocked`—more deliberate that it is part of the slashing lifecycle.